OpenRA.Mods.Common/Traits/World/MPStartLocations.cs

```diff
@@ -58,13 +58,13 @@ IEnumerable<LobbyOption> ILobbyOptions.LobbyOptions(Ruleset rules)
 		}
 	}
 
-	public class MPStartLocations : IWorldLoaded, INotifyCreated
+	public class MPStartLocations : IWorldLoaded, INotifyCreated, IAssignSpawnPoints
 	{
 		readonly MPStartLocationsInfo info;
-
-		public readonly Dictionary<Player, CPos> Start = new Dictionary<Player, CPos>();
-
 		bool separateTeamSpawns;
+		CPos[] spawnLocations;
+		List<int> availableSpawnPoints;
+		Dictionary<int, Session.Client> occupiedSpawnPoints;
```

Comment:
This can be made readonly since the dictionary creation itself doesn't depend on `Created`.